src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,55 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getKindex(key);
+
+        if (index >= 0) {
+            values[index] = value;
+        } else {
+            if (size >= MAX_SIZE) {
+                // Storage is full, cannot add more elements
+                throw new IllegalStateException("Storage is full");
+            }
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
```

Comment:
Good job, but handling storage overflow is out of the scope of this task :) Also, remove comments before committing 
```suggestion
        if (index >= 0) {
            values[index] = value;
        } else {
            keys[size] = key;
            values[size++] = value;
        }
```